luigi/hdfs.py

```diff
@@ -105,6 +105,51 @@ def list_path(path):
         return [path, ]
     return [str(path), ]
 
+def is_dangerous_rm_path(path):
+    """ Finds out if the path is too dangerous to remove.
+    * empty path
+    * / root path
+    * one level deep root path, i.e. /etc or /opt
+    :return boolean, True if too dangerous! Relative path is always ok
+
+    >>> is_dangerous_rm_path(None)
+    True
+    >>> is_dangerous_rm_path('')
+    True
+    >>> is_dangerous_rm_path(' ')
+    True
+    >>> is_dangerous_rm_path('/opt')
+    True
+    >>> try:
+    ...     is_dangerous_rm_path('//')
+    ... except ValueError:
+    ...     pass
+    >>> is_dangerous_rm_path('/opt/')
+    True
+    >>> is_dangerous_rm_path('/opt/foo/bar')
+    False
+    >>> is_dangerous_rm_path('my_relative')
+    False
+    """
+    if not path:
+        return True
+
+    if '//' in path:
+        raise ValueError('Funny path, double slash: %s' % path)
+
+    path = path.strip()
+
+    if path.endswith('/'):
+        path = path[:-1]
```

Comment:
Hmm, I actually never realized strip can take a parameter. Learn something new everyday, and something so basic!
strip('/' would probably then strip the leading '/', thereby making the path appear relative, and by the intention of the function, we allow users to delete relative files with impunity.
I would have to check to see if the path appears relative, but leading spaces may interfere with that check.